operator/docs/enhancements/short_lived_tokens_authentication.md

```diff
@@ -0,0 +1,355 @@
+- Google Cloud Platform: [Workload Identity Federation](https://cloud.google.com/iam/docs/workload-identity-federation)
+
+## Motivation
+
+Building upon the fact that short-lived token authentication is supported by all three object storage clients in Loki (Azure, AWS, GAP), this proposal's main focus is to automate the administration effort to run LokiStack using such an authentication workflow for object storage access.
+
+### Goals
+
+- The LokiStack administrator provides only minimal but valid object storage secrets that enable short-lived token authentication for all three major cloud providers.
+- The LokiStack administrator is not required to expose credentials/identity-tokens/etc. to the cloud provider when operating in that workflow.
+
+### Non-Goals
+
+- Provide auxiliary and comprehensive tooling to automate preparing the cloud provider IAM resources for short-lived token authentication.
+
+## Proposal
+
+The following proposal describes the required changes and validations for each object storage secret for each major cloud provider (Azure, AWS, GCP).
+
+### API Extensions
+
+The present proposal introduces a new optional field in the `ObjectStorageSecretSpec` namely `CredentialsMode`. If the user does not provide a value the `CredentialsMode` is automatically detected either from the provided secret fields or the operator environment variables (latter applies only in OpenShift cluters.). The selected or detected `CredentialsMode` is populated in addition in the `status.storage.credentialMode` field.
+
+```go
+/ CredentialMode represents the type of authentication used for accessing the object storage.
+//
+// +kubebuilder:validation:Enum=static;token;managed
+type CredentialMode string
+
+const (
+    // CredentialModeStatic represents the usage of static, long-lived credentials stored in a Secret.
+    // This is the default authentication mode and available for all supported object storage types.
+    CredentialModeStatic CredentialMode = "static"
+    // CredentialModeToken represents the usage of short-lived tokens retrieved from a credential source.
+    // In this mode the static configuration does not contain credentials needed for the object storage.
+    // Instead, they are generated during runtime using a service, which allows for shorter-lived credentials and
+    // much more granular control. This authentication mode is not supported for all object storage types.
+    CredentialModeToken CredentialMode = "token"
+    // CredentialModeManaged represents the usage of short-lived tokens retrieved from a credential source.
+    // This mode is similar to CredentialModeToken,but instead of having a user-configured credential source,
+    // it is configured by the environment, for example the Cloud Credential Operator in OpenShift.
+    // This mode is only supported for certain object storage types in certain runtime environments.
+    CredentialModeManaged CredentialMode = "managed"
```

Comment:
```suggestion
  // CredentialModeTokenCCO represents the usage of short-lived tokens retrieved from a credential source.
  // This mode is similar to CredentialModeToken, but instead of having a user-configured credential source,
  // it is configured by the environment and the operator relies on the Cloud Credential Operator to provide
  // a secret. This mode is only supported for certain object storage types in certain runtime environments.
  CredentialModeTokenCCO CredentialMode = "token-cco"
```